Makefile

```diff
@@ -76,62 +76,33 @@ ifeq ($(NO_DISCOVER),true)
 	BASE_CMD += --nodiscover
 endif
 
-# Build suburl strings for slice specific subclient groups
-# WARNING: Only connect to dom/sub clients over a trusted network.
-ifeq ($(REGION),2)
-	PRIME_SUBS += ,,ws://127.0.0.1:$(REGION_$(REGION)_PORT_WS)
-endif
-ifeq ($(REGION),1)
-	PRIME_SUBS += ,ws://127.0.0.1:$(REGION_$(REGION)_PORT_WS),
-endif
-ifeq ($(REGION),0)
-	PRIME_SUBS += ws://127.0.0.1:$(REGION_$(REGION)_PORT_WS),,
-endif
-ifeq ($(ZONE),2)
-	REGION_SUBS =,,ws://127.0.0.1:$(ZONE_$(REGION)_$(ZONE)_PORT_WS)
-endif
-ifeq ($(ZONE),1)
-	REGION_SUBS =,ws://127.0.0.1:$(ZONE_$(REGION)_$(ZONE)_PORT_WS),
-endif
-ifeq ($(ZONE),0)
-	REGION_SUBS =ws://127.0.0.1:$(ZONE_$(REGION)_$(ZONE)_PORT_WS),,
-endif
+build_sequence = $(if $(filter $1,$2),,$1 $(call build_sequence,$(shell expr $1 + 1),$2))
+
+# Generate the sequence outside of any target
+WIDTH_SEQ := $(call build_sequence,0,$(WIDTH))
+
+# Step 2: Execute the commands in the run target's recipe
+# Command for the prime node
+PRIME_CMD := nohup $(BASE_CMD) --port $(PRIME_PORT_TCP) --http.port $(PRIME_PORT_HTTP) --ws.port $(PRIME_PORT_WS) --sub.urls $(PRIME_SUB_URLS) >> nodelogs/prime.log 2>&1 &
+
+# Template for the region nodes
+define REGION_CMD
+nohup $(BASE_CMD) --port $(REGION_$(1)_PORT_TCP) --http.port $(REGION_$(1)_PORT_HTTP) --ws.port $(REGION_$(1)_PORT_WS) --dom.url $(REGION_$(1)_DOM_URL):$(PRIME_PORT_WS) --sub.urls $(REGION_$(1)_SUB_URLS) --region $(1) >> nodelogs/region-$(1).log 2>&1 &
```

Comment:
why is this $(1) hard coded ?

Reply:
Function index (first parameter is the region index)